src/Io/MultipartParser.php

```diff
@@ -101,7 +101,6 @@ public function parse(ServerRequestInterface $request)
         $this->parseBody('--' . $matches[1], (string)$request->getBody());
 
         $request = $this->request;
-        $this->request = null;
```

Comment:
ServerRequestInterface::$request does not accept null

Reply:
> ServerRequestInterface::$request does not accept null

This is about `MultipartParser->$request` which can be null and not related to parameter passing?

Reply:
Good catch, but with this change applied this instance will now keep a reference to the request and will thus avoid garbage collecting this instance until this method is called the next time.

How about using a `?ServerRequestInterface` type hint for this property instead? This also makes sense because the constructor does not assign an instance to this property.